tests/python/unit_tests/test_friends.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2025 NWChemEx-Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from importlib.util import find_spec
+
+from friendzone.friends import friends, is_friend_enabled
+
+
+class TestFriends(unittest.TestCase):
+    def test_friends_list(self):
+        expected_friends = ["ase", "nwchem"]
+        actual_friends = friends()
+        self.assertCountEqual(actual_friends, expected_friends)
+
+    def test_is_friend_enabled(self):
+        # For known friends, the result should match the system state
+        for friend in friends():
+            enabled = is_friend_enabled(friend)
+            module_found = find_spec(friend) is not None
+            self.assertEqual(enabled, module_found)
+        # For unknown friends, the result should be False
+        self.assertFalse(is_friend_enabled("non_existent_friend"))
```

Comment:
I'm not sure about this test. I'd rather see this split into separate tests for each friend, or at least one check per "type" of friend (Python package, installed binary, etc.). Right now we only have one "type", but NWChem would fit under "installed binary" if we choose to fully check for it in the future.

These tests would likely need to be run separately in different virtual environments, which, I admit, would complicate the CI a bit. I'm not sure what the best approach would be with unittest, but [pytest marks](https://docs.pytest.org/en/stable/how-to/mark.html#mark) may be useful here, specifically [custom markers](https://docs.pytest.org/en/stable/example/markers.html).

Reply:
I'm realizing that a greater amount of change is needed for checking (and also specifying) our "friends." This test will mostly likely be updated with pending changes I'm working on.

Regarding CI changes to enumerate possible dependency states, I think that's outside of the scope of what needs to be done here. But that we should open an issue for it.